src/frontend/js/widgets/Dashboard/utils/teacherRouteMessages.tsx

```diff
@@ -6,109 +6,22 @@ import { defineMessages } from 'react-intl';
 export enum TeacherDashboardPaths {
   ROOT = '/teacher',
   TEACHER_COURSES = '/teacher/courses',
-  ORGANIZATION = '/teacher/organizations/{organizationId}',
-  ORGANIZATION_CONTRACTS = '/teacher/organizations/{organizationId}/contracts',
-  ORGANIZATION_COURSES = '/teacher/organizations/{organizationId}/courses',
-  ORGANIZATION_PRODUCT = '/teacher/organizations/{organizationId}/courses/{courseId}/products/{courseProductRelationId}',
-  ORGANIZATION_COURSE_CONTRACTS = '/teacher/organizations/{organizationId}/courses/{courseId}/contracts',
-  ORGANIZATION_PRODUCT_CONTRACTS = '/teacher/organizations/{organizationId}/courses/{courseId}/products/{courseProductRelationId}/contracts',
-  ORGANIZATION_COURSE_PRODUCT_LEARNER_LIST = '/teacher/organizations/{organizationId}/courses/{courseId}/products/{courseProductRelationId}/learners',
-  ORGANIZATION_COURSE_GENERAL_INFORMATION = '/teacher/organizations/{organizationId}/courses/{courseId}/information',
-  COURSE = '/teacher/courses/{courseId}',
-  COURSE_GENERAL_INFORMATION = '/teacher/courses/{courseId}/information',
-  COURSE_PRODUCT = '/teacher/courses/{courseId}/products/{courseProductRelationId}',
-  COURSE_PRODUCT_LEARNER_LIST = '/teacher/courses/{courseId}/products/{courseProductRelationId}/learners',
-  COURSE_CONTRACTS = '/teacher/courses/{courseId}/contracts',
-  COURSE_PRODUCT_CONTRACTS = '/teacher/courses/{courseId}/products/{courseProductRelationId}/contracts',
+  ORGANIZATION = '/teacher/organizations/:organizationId',
+  ORGANIZATION_CONTRACTS = '/teacher/organizations/:organizationId/contracts',
+  ORGANIZATION_COURSES = '/teacher/organizations/:organizationId/courses',
+  ORGANIZATION_PRODUCT = '/teacher/organizations/:organizationId/courses/:courseId/products/:courseProductRelationId',
+  ORGANIZATION_COURSE_CONTRACTS = '/teacher/organizations/:organizationId/courses/:courseId/contracts',
+  ORGANIZATION_PRODUCT_CONTRACTS = '/teacher/organizations/:organizationId/courses/:courseId/products/:courseProductRelationId/contracts',
+  ORGANIZATION_COURSE_PRODUCT_LEARNER_LIST = '/teacher/organizations/:organizationId/courses/:courseId/products/:courseProductRelationId/learners',
+  ORGANIZATION_COURSE_GENERAL_INFORMATION = '/teacher/organizations/:organizationId/courses/:courseId/information',
+  COURSE = '/teacher/courses/:courseId',
+  COURSE_GENERAL_INFORMATION = '/teacher/courses/:courseId/information',
+  COURSE_PRODUCT = '/teacher/courses/:courseId/products/:courseProductRelationId',
+  COURSE_PRODUCT_LEARNER_LIST = '/teacher/courses/:courseId/products/:courseProductRelationId/learners',
+  COURSE_CONTRACTS = '/teacher/courses/:courseId/contracts',
+  COURSE_PRODUCT_CONTRACTS = '/teacher/courses/:courseId/products/:courseProductRelationId/contracts',
```

Comment:
Now, should we use template string to prevent to repeat paths and reduce the risk of typo ?

```js
  ORGANIZATION = '/teacher/organizations/:organizationId',
  ORGANIZATION_CONTRACTS = `${ORGANIZATION}/contracts`,
```

Reply:
Tbh I think having plain string gives a faster way to read the route full url, improving DX. But .. that's only my %, let's talk about it during the next daily?

Reply:
I agree with @jbpenrath suggestion

Reply:
The downside of this proposal is that broken string are harder to grep. Also a grep of `COURSE_PRODUCT` to found all usage of this route will give false postive usage like 
```ts
const COURSE_PRODUCT_LEARNER_LIST = `${COURSE_PRODUCT}/learners`
```
The only advantage is that we avoid typo.

At first i like the idea but it seems not needed at the end.